src/engine/render/texture.rs

```diff
@@ -43,16 +56,18 @@ struct TextureGLState {
 }
 
 impl Texture {
-    pub fn new_empty(width: u32, height: u32) -> Rc<Self> {
+    pub fn new_render_texture(width: u32, height: u32) -> Rc<Self> {
         Rc::new(Texture {
             filtering: TextureFiltering::Linear,
-            img: Resource::new(ImageBuffer::new(width, height)),
             gl_state: RefCell::new(None),
+            kind: TextureKind::RenderTexture(RenderTexture {

```

Comment:
I think there's an issue here. The RenderTexture, once generated, has to be used as an Image texture. You need at least a function to turn a renderTexture into a standard texture.

Reply:
You mean render it as a normal texture or get the image buffer itself out to modify directly?

Reply:
render it as a normal texture. The steps are :
- create an empty texture (that would be a RenderTexture kind)
- associated it with a framebuffer and bind the framebuffer
- render the scene (to the texture)
- then render something else using the RenderTexture as a standard texture (simply use this texture as a shader parameter). Now, this texture is an Image kind.